lib/engine/depot.rb

```diff
@@ -89,9 +89,20 @@ def discountable_trains_for(corporation)
           discounted_price = discount_train.price(train)
           next if discount_train.price == discounted_price
 
-          [train, discount_train, discounted_price]
+          name = discount_train.name
+          discount_info = [[train, discount_train, name, discounted_price]]
+
+          # Add variants if any - they have same discount as base version
+          discount_train.variants.each do |_, v|
+            next if v[:name] == name
+
+            price = v[:price] - (discount_train.price - discounted_price)
+            discount_info << [train, discount_train, v[:name], price]
+          end
+
+          discount_info
         end.compact
-      end
+      end.flatten(1)
```

Comment:
shouldn't flat_map just work?

Reply:
This is me being unused to Ruby... you are correct, flat_map gives the same result.

Consider line 93-100: is there a better way to inject extra elements during a map operation?
Normally the map [t1, t2, t3] would result in [[t1, _, _], [t2, _, _], [t3, _, _]] but I wanted to insert a [t1-variant, _, _] extra. The code I wrote do this but it felt a bit clunky...